File: Oidc.AuthorizationServer/Models/PostLogoutRedirectUri.cs
using FluentNHibernate.Data;
using lvl.Ontology;
using lvl.Ontology.Authorization;
using lvl.Ontology.Conventions;
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.Models
{

    /// <summary>
    ///     Determines allowed URIs to redirect to after logout.
    /// </summary>
    [Schema("oidc")]
    [HiddenFromApi]
    public class PostLogoutRedirectUri : Entity, IAggregateScope<ClientEntity>
    {
        /// <summary>
        ///     A url which is allowed to be called after logout.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/post-logout-redirect-uri.ts
import { ClientEntity } from './client-entity';
import { Entity, IAggregateScope, Required } from '@lvl/front-end';

export class PostLogoutRedirectUri extends Entity implements IAggregateScope<ClientEntity> {
    @Required() public name: string;

    constructor(options?: IPostLogoutRedirectUriOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface IPostLogoutRedirectUriOptions {
    name?: string;
    id?: number;
}
